test/FunctionalTests/Microsoft.OData.Core.Tests/ODataTextStreamReaderTests.cs

```diff
@@ -0,0 +1,71 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataTextStreamReaderTests.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.OData.Tests
+{
+    /// <summary>
+    /// Unit tests for <see cref="ODataTextStreamReader"/>.
+    /// </summary>
+    public class ODataTextStreamReaderTests
+    {
+        private MemoryStream stream;
+        private const int maxLength = 14;
+
+        public ODataTextStreamReaderTests()
+        {
+            this.stream = new MemoryStream(Encoding.UTF8.GetBytes("The quick brown fox jumps over the lazy dog."));
+            this.stream.Position = 16;

```

Comment:
Why are we setting the Position as 16?
I thought to read a stream, the position need to be at 0.

Reply:
@KenitoInc For the fun of it. I just wanted to start to read from a random position instead of the typical 0

Reply:
It might be good to have a test for both

Reply:
Maybe randomize it from 0 to length :)